drafts/draft-0005.md

```diff
@@ -0,0 +1,188 @@
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "statusHref": "https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036",
+  "extras": {
+    "isStellarGuard": true,
+    "url": "https://stellarguard.me/transactions/c585ceee-b1b9-4009-aa37-b8544346a036"
+  }
+}
+```
+
+#### Status Endpoint
+
+This will be used to query for the status of a multisig transaction.
+
+##### Request
+
+`GET MULTISIG_ENDPOINT>/:id`
+
+Request Path Parameters:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. The id used to look up the status, which was returned in the submit transaction response.
+
+Example:
+
+```
+GET https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036
+```
+
+##### Response
+
+On success, the endpoint should return a `200 OK` HTTP status code and a JSON-encoded object.
+
+Response fields:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. An identifier for the multisig request that is generated by the multisig service.
+`status` | string | Required. An string indicating the status of the multisig request which can have one of the following values:<br><br>- `pending` - The request requires more signatures<br>- `success` - The transaction has been fully signed<br>- `failed` - There was an error with the transaction and it could not be submitted successfully
+`uri` | string | Required. The SEP-0007 URI that was submitted to the service
+`signers`| string[] | Required. An array of public keys that have signed the transaction. 
+
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "status": "pending",
+  "uri": "web+stellar:tx?xdr=AAAAAHFd0%2BHQV5u%2FY%2FfM3%2BVelUr1IWwSFL8CUDIAUudUdD4MAAAAZAADyI8AAAADAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAcV3T4dBXm79j98zf5V6VSvUhbBIUvwJQMgBS51R0PgwAAAAAAAAAAACYloAAAAAAAAAAAVR0PgwAAABAEw8ODG0iixkbAHg1aJATAnZS2531PhGauuSvFDad2WxHKzIenUNbc7K5mGiSpe5jvqe19OQCbNFuBjqN11jfBw%3D%3D",
+  "signers": ["GBYV3U7B2BLZXP3D67GN7ZK6SVFPKILMCIKL6ASQGIAFFZ2UOQ7AZB5V"]
+}
+```
+
+####  CORS headers
+
+In order to comply with browser cross-origin access policies, the service should provide wildcard CORS response HTTP header. The following HTTP header must be set for the API endpoints:
+
+```
+Access-Control-Allow-Origin: *
+```
+
+### Multisig Transaction Flow
+
+1. Determine whether a transaction requires additional signatures in order to be valid.
+1. If the transaction requires more signatures, look up the data entry with the key `multisig.domain` on the transaction source account.
+1. Resolve the `stellar.toml` associated with the account's `multisig.domain` entry and return the `MULTISIG_ENDPOINT` field.
+1. Convert the transaction to a SEP-0007 style URI, setting any additional parameters such as `callback`.
+1. Before submitting the transaction, it should contain at least one valid signature.
```

Comment:
As said, this should be vendor specific

Reply:
Then nothing prevents the coordinator to use them if he consider then necessary. I would put the question the other way: what's the benefit of forbidding coordinator to accept unsigned transaction - maybe they authenticated the user by some other means. How does it matter to us?

I don't think I would actually send unsigned transaction but I certainly wouldn't forbid possibilities just because I don't see how it may be put at use in the future.

Reply:
I agree with your goal of keeeping things flexible here. The questions I’ve posed above are important to be answered before we can allow offering both signed and unsigned options in the procol, otherwise the protocol will be _broken_ because wallets will not know whether the service accepts signed or unsigned XDRs.

Anyhow, since we don’t have a use case right now then it does not make sense to bloat up the SEP, which only makes it more expensive to implement. We can always extend the SEP if we have a real use case for this later.

Reply:
Your answers:

1. Wallets should not send unsigned transaction to coordinator.
2. Example use case: a smart-contract coordinator that accept external inputs from a (likely authenticated) non-signing tier; Or more generally signature coordinator that can be configured to accept external tx request for authenticated tiers.

Reply:
1. Ok, so we are in agreement that all XDRs that are sent to the multisig service via this protocol should be **signed**?
2. This is inconsistent with ^^your point one above^^. If you have a use-case for sending XDRs that are not signed to a multisig service then why do you say that `wallets should not send unsigned transactions to coordinator` in your point above?

In your example use case, what is an example of an authenticated tier? How will it be authenticated? Please provide more details on this.

Reply:
I think we can resolve this point by keeping it this way for now, because you guys are right that in general it's the safest thing to do. If coordinators have to deviate from this rule to implement new features they will simply do so - no harm here.